pkg/agent/bgp/engine/gobgp.go

```diff
@@ -0,0 +1,295 @@
+// Copyright 2024 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import (
+	"context"
+	"fmt"
+	"net/netip"
+	"time"
+
+	gobgpapi "github.com/osrg/gobgp/v3/api"
+	"github.com/osrg/gobgp/v3/pkg/server"
+	"google.golang.org/protobuf/types/known/anypb"
+	"k8s.io/utils/net"
+)
+
+const (
+	ipv4AllZero = "0.0.0.0"
+	ipv6AllZero = "::"
+)
+
+type GoBGPServer struct {
+	server       *server.BgpServer
+	globalConfig *gobgpapi.Global
+}
+
+func NewGoBGPServer(globalConfig *GlobalConfig, extraLogFields map[string]interface{}) (Interface, error) {
```

Comment:
Since the BGP logs are in `gobgp.log`, I thought that it would be more convenient for users to debug when there are multiple `BGPPolicy` applied to a Node after rebooting (the effective one might be changed after rebooting). 

> There is only one effective BGPPolicy and only one ASN, why isn't it enough we just log the policy and ASN when receiving it?

That's fine with me. The current implementation is too redundant for logging.